### ISSUE TITLE
Consider liquibase

### ISSUE THREAD

**Issue by hohwille** [devonfw/devon4j#303]
We are aiming to propose a main default suggestion for every problem and therefore have our [key-principles](https://github.com/devonfw/devon4j/blob/develop/documentation/architecture.asciidoc#key-principles). Following KISS we have choosen [flywaydb](https://flywaydb.org/) for DB versioning as it is simple and you typically just put your SQL files as you would manually run them on your DB.

A famous alternative to flyway is [liquibase](https://www.liquibase.org/) that allows some DB abstraction. This is typically not needed as most projects have one supported DB to run on. However, flyway removed some features as support for specific Oracle versions and caused some severe issues to our projects in production forcing them to buy a commercial license. While it is OK for commecial projects to buy a commercial license, it was surprising that an OSS edition removed support in a newer version. In this case several users requested liquibase support. As we do not want to drop and replace flyway with liquibase we should consider adding liquibase as a supported option.

To summarize this story:
* [ ] add a config option to toggle to liquibase in our [app template](https://github.com/devonfw/devon4j/tree/develop/templates/server).
* [ ] extend our [DB migration guide](https://github.com/devonfw/devon4j/blob/develop/documentation/guide-database-migration.asciidoc) to support liquibase

**Comment by hohwille** [comment]
Rationale that can be added to the documentation:
```
[ERROR] Failed to execute goal org.flywaydb:flyway-maven-plugin:7.5.3:clean (default-cli) on project demo-core: org.flywaydb.core.internal.license.FlywayTeamsUpgradeRequiredException: Flyway Teams Edition upgrade required: oracle.sqlplus is not supported by Flyway Community Edition. -> [Help 1]
```
or
```
org.flywaydb.core.internal.license.FlywayEnterpriseUpgradeRequiredException: Flyway Enterprise Edition or Oracle upgrade required: Oracle 11.2 is past regular support by Oracle and no longer supported by Flyway Community Edition, but still supported by Flyway Enterprise Edition.
```

**Comment by baumeister25** [comment]
@hohwille @dixyushi 
Is there any progress on this story? Do you need support?
